qt/python/mantidqtinterfaces/test/Muon/utilities/thread_model_test.py

```diff
@@ -84,6 +76,13 @@ def mock_model(self):
         model.output = mock.Mock()
         return model
 
+    def wait_for_thread(self, thread_model):
+        if thread_model and thread_model.worker:
+            while thread_model.worker.is_alive():
+                QApplication.sendPostedEvents()
+                time.sleep(0.1)
+            QApplication.sendPostedEvents()
+
     # ------------------------------------------------------------------------------------------------------------------
     # TESTS
```

Comment:
Do we need to add any new tests now we're using async threading?

Reply:
With all the test that utalise the threads passing without having to change them beyond the wait_for_thread the existing tests should be sufficent.